projects/keycloak-angular/src/lib/core/interceptors/keycloak-bearer-fn.interceptor.ts

```diff
@@ -0,0 +1,109 @@
+import {
+  HttpEvent,
+  HttpHandlerFn,
+  HttpInterceptorFn,
+  HttpRequest
+} from '@angular/common/http';
+import { inject } from '@angular/core';
+import { KeycloakService } from '../services/keycloak.service';
+import { ExcludedUrlRegex } from '../interfaces/keycloak-options';
+import { Observable, combineLatest, from, mergeMap, of } from 'rxjs';
+
+export const keycloakBearerFnInterceptor: HttpInterceptorFn = (req, next) => {
```

Comment:
Hi @apoveda25,

I found your PR and would greatly appreciate it if the maintainers could merge it into the library - thanks for contributing it!

I have just one minor note from my side: I would stick with the "naming scheme" from Angular and call the interceptor `keycloakBearerInterceptorFn` instead as this aligns with the [official typings](https://angular.dev/api/common/http/HttpInterceptorFn) and documentation of Angular...

Hope this PR makes it to the main branch soon. 🙂 

Best regards! 🎉